libpromises/conversion.c

```diff
@@ -509,7 +509,7 @@ bool DoubleFromString(const char *s, double *value_out)
 
     remainder[0] = '\0';
 
-    sscanf(s, "%lf%c%s", &a, &c, remainder);
+    sscanf(s, "%lf%c%4095s", &a, &c, remainder);
```

Comment:
Cherry-picking since it's not worse than before, I'll repeat though that there is no meaning in having 4K memory overhead. Reading "remainder" in a `char` variable would be enough.